models/optional_modules/consent/snowplow_web_consent_versions.sql

```diff
@@ -0,0 +1,91 @@
+{{
+  config(
+    materialized='incremental',
+    unique_key='consent_version',
+    sort = 'version_start_tstamp',
+    dist = 'consent_version',
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+
+with consent_versions as (
+
+  select
+    consent_version,
+    min(derived_tstamp) as version_start_tstamp,
+    consent_scopes,
+    consent_url,
+    domains_applied,
+    derived_tstamp as last_processed_event_tstamp,
+    row_number() over(partition by consent_version order by derived_tstamp desc) as row_count
+
+  from {{ ref('snowplow_web_consent_log') }}
+
+  where event_name <> 'cmp_visible' and event_type = 'allow_all'
+
+  -- enough time has passed since the event happened (derived_tstamp) to be able to process it as a whole (please bear in mind the late arriving data)
+  and derived_tstamp < cast({{ dbt_utils.dateadd('hour', var("snowplow__consent_late_arriving_window", 10), dbt_utils.current_timestamp_in_utc() ) }} as {{ dbt_utils.type_timestamp() }})
+
```

Comment:
❓ Does this mean that if they make their `allow_all` event we ask them to at the start they won't see it in this table for 10 hours?

Reply:
Unfortunately yes. Another' ugly' bit to work around tracking. It will be emphasised in the tracking implementation manual, probably with this in mind to have this a day before the go live or sth.